atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/ValidatingQueryRewritingKeyValueService.java

```diff
@@ -97,10 +102,11 @@ public void createTables(Map<TableReference, byte[]> tableRefToTableMetadata) {
     protected void sanityCheckTableName(TableReference tableRef) {
         String tableName = tableRef.getQualifiedName();
         Validate.isTrue(
-                (!tableName.startsWith("_") && tableName.contains("."))
+                (!tableName.startsWith("_") && (tableName.contains(".")))
+                        && ALLOWED_TABLE_NAMES.matcher(tableName).matches()
```

Comment:
nit: would prefer parentheses to make the operator precedence clearer (as with the existing && check)